test/functional/inspec_exec_test.rb

```diff
@@ -4,13 +4,37 @@
   include FunctionalHelper
   let(:looks_like_a_stacktrace) { %r{lib/inspec/.+\.rb:\d+:in} }
 
+  attr_accessor :out
+
+  def inspec(commandline, prefix = nil)
```

Comment:
As described in the commits, this refactoring of the test file cleans up both the IO redundancy throughout the entire file and also reorders the exit code checks to be last so that failing tests are actually debuggable.

Putting this in a separate PR doesn't make sense in this instance. Having 4 PRs with single commits that must be stacked in order to get the real work done instead of a single PR with 4 commits that all do small(ish) unrelated but overlapping things only serves to slow us down and add bureaucratic process where none is needed.

Reply:
Putting this in a separate PR doesn't make sense in this instance. Having 4 PRs with single commits that must be stacked in order to get the real work done instead of a single PR with 4 commits that all do small(ish) unrelated but overlapping things only serves to slow us down and add bureaucratic process where none is needed.